language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/TypeTestExpressionNodeContext.java

```diff
@@ -97,50 +97,55 @@ public TypeTestExpressionNodeContext() {
             //check for intersection as well.
             return completionItems;
         }
-        completionItems = typeReferences.stream().filter(typeRef -> isQualifiedTypeReference(context, typeRef))
-                .map(CommonUtil::getRawType)
-                .map(typeRef -> {
-                    String symbolRef = getQualifiedSymbolReference(typeRef);
-                    return new SymbolCompletionItem(context, typeRef,
-                            TypeCompletionItemBuilder.build(typeRef, symbolRef));
-                }).collect(Collectors.toList());
+        typeReferences.stream()
+                .forEach(typeRef -> {
+
+                    Optional<String> symbolRef = getQualifiedSymbolReference(context, typeRef);
+                    if (symbolRef.isPresent()) {
+                        TypeSymbol rawType = CommonUtil.getRawType(typeRef);
+                        completionItems.add(new SymbolCompletionItem(context, rawType,
+                                TypeCompletionItemBuilder.build(rawType, symbolRef.get())));
+                    }
+                });
         return completionItems;
     }
 
-    private String getQualifiedSymbolReference(TypeSymbol typeSymbol) {
-        ModuleID moduleID = typeSymbol.getModule().get().id();
-        String moduleName = moduleID.moduleName();
+    private Optional<String> getQualifiedSymbolReference(BallerinaCompletionContext context,
+                                                         TypeSymbol typeSymbol) {
+
+        Optional<ModuleSymbol> module = typeSymbol.getModule();
+        Optional<Module> currentModule = context.currentModule();
+        Optional<String> name = typeSymbol.getName();
+
+        if (typeSymbol.typeKind() != TypeDescKind.TYPE_REFERENCE
+                && module.isEmpty() || name.isEmpty()
```

Comment:
Instead of this method, what if we try to use `CommonUtil.getModifiedTypeName` util API? Are there any uncovered cases with that usage?

Reply:
+1 for using the said method. However we have to check if a particular typesymbol is not from the current  module. Therefore, first we need to check if the is not from the current module and then use the `CommonUtil.getModifiedTypeName` to generate the type name. WDYT?

Reply:
+1. Let's make a note to improve such filtering logics in the future.